libs/features/mas/src/aem-fragment.js

```diff
@@ -104,61 +115,91 @@ export class AemFragment extends HTMLElement {
     connectedCallback() {
         if (this.#fetchPromise) return;
         this.#service = getService(this);
-        this.#log = this.#service.log.module(AEM_FRAGMENT_TAG_NAME);
-        this.#startMark = `${AEM_FRAGMENT_TAG_NAME}:${this.#fragmentId}${MARK_START_SUFFIX}`;
-        performance.mark(this.#startMark);
-        if (!this.#fragmentId) {
-            this.#fail({ message: 'Missing fragment id' });
+        this.#log = this.#service.log.module(
+            `${AEM_FRAGMENT_TAG_NAME}[${this.#fragmentId}]`,
+        );
+        // TODO get rid of the special case for #
+        if (!this.#fragmentId || this.#fragmentId === '#') {
+            this.#fail('Missing fragment id');
             return;
         }
         this.refresh(false);
     }
 
+    get fetchInfo() {
+        return Object.fromEntries(
+            Object.entries(this.#fetchInfo)
+                .filter(([key, value]) => value != undefined)
+                .map(([key, value]) => [`aem-fragment:${key}`, value]),
+        );
+    }
+
     /**
      * Get fragment by ID
      * @param {string} endpoint url to fetch fragment from
      * @param {string} id fragment id
      * @returns {Promise<Object>} the raw fragment item
      */
-    async getFragmentById(endpoint, id, startMark) {
-        const measureName = `${AEM_FRAGMENT_TAG_NAME}:${id}${MARK_DURATION_SUFFIX}`;
+    async #getFragmentById(endpoint) {
+        this.#fetchCount++;
+        const markPrefix = `${AEM_FRAGMENT_TAG_NAME}:${this.#fragmentId}:${this.#fetchCount}`;
+        const startMarkName = `${markPrefix}${MARK_START_SUFFIX}`;
+        const measureName = `${markPrefix}${MARK_DURATION_SUFFIX}`;
+        performance.mark(startMarkName);
         let response;
         try {
+            this.#fetchInfo.stale = false;
             response = await masFetch(endpoint, {
                 cache: 'default',
                 credentials: 'omit',
             });
+            this.#applyHeaders(response);
+            this.#fetchInfo.status = response?.status;
+            const { startTime, duration } = performance.measure(
+                measureName,
+                startMarkName,
+            );
+            this.#fetchInfo.startTime = parseFloat(startTime.toFixed(2));
+            this.#fetchInfo.duration = parseFloat(duration.toFixed(2));
```

Comment:
you could just get string versions, no? we know from perf api that those will be numbers

```suggestion
            this.#fetchInfo.startTime = startTime.toFixed(2);
            this.#fetchInfo.duration = duration.toFixed(2);
```

Reply:
I don't want Numbers to be quoted in the logs, `toFixed` converts numbers to digits.